celery_utils/logged_task.py

```diff
@@ -36,6 +36,14 @@ def on_retry(self, exc, task_id, args, kwargs, einfo):
         """
         Capture the exception that caused the task to be retried, if any.
         """
-        super(LoggedTask, self).on_retry(exc, task_id, args, kwargs, einfo)
         if exc is not None:
             log.warning('[{}] retried due to {}'.format(task_id, einfo.traceback))
+        super(LoggedTask, self).on_retry(exc, task_id, args, kwargs, einfo)
+
+    def on_failure(self, exc, task_id, args, kwargs, einfo):
```

Comment:
Some more investigation results: celery >= 4.0 does the logging listed above, but celery 3.1.18 (which we're using), does not. We could, however, subscribe to `celery.signals.task_retry` and `celery.signals.task_failure` to log globally on those events, rather than needing to decorate every task.